File: verisigil/verisigil/verisigil/verisigil/verisigil/js/src/examples/python/quickstart.py
"""
VeriSigil AI — Quick Start Example
Run: python examples/python/quickstart.py
"""

from verisigil import VeriSigil

vs = VeriSigil(api_key="demo")

print("=" * 55)
print("  VeriSigil AI — Demo")
print("=" * 55)

# Issue passport
print("\n🔐 Issuing passport...")
passport = vs.issue_passport(
    agent_name="financial-analysis-agent",
    owner="team@example.com",
    framework="langchain",
)
print(f"   ✅ Issued!")
print(f"   DID:          {passport.did}")
print(f"   Trust Score:  {passport.trust_score}")
print(f"   EU Risk:      {passport.eu_risk_class.value}")
print(f"   Status:       {passport.status.value}")
print(f"   Expires:      {passport.expires_at.strftime('%Y-%m-%d')}")

# Verify
print("\n🛡️  Verifying...")
print(f"   {'✅ Verified' if vs.verify(passport.agent_id) else '⛔ NOT verified'}")

# Scan
print("\n🔍 Security scan...")
code = """
api_key = "sk-1234"
result = eval(user_input)
"""
results = vs.scan(code)
print(f"   Threats: {results['threat_count']}")
for t in results["threats"]:
    icon = "🔴" if t["severity"] == "HIGH" else "🟡"
    print(f"   {icon} [{t['severity']}] Line {t['line']}: {t['description']}")

# Compliance
print("\n⚖️  Compliance...")
c = vs.check_compliance(passport.agent_id)
print(f"   EU AI Act: {'✅' if c['eu_ai_act']['compliant'] else '❌'}")
print(f"   GDPR:      {'✅' if c['gdpr']['compliant'] else '❌'}")

print("\n" + "=" * 55)
print("  🚀 verisigilai.com  |  info@verisigilai.com")
print("=" * 55)

File: examples/javascript/quickstart.js
/**
 * VeriSigil AI — JavaScript Quick Start
 * Run: node examples/javascript/quickstart.js
 */

'use strict';

const { VeriSigil } = require('../../js/src/index');

async function main() {
  console.log('='.repeat(55));
  console.log('  VeriSigil AI — JavaScript Demo');
  console.log('='.repeat(55));

  const vs = new VeriSigil({ apiKey: 'demo' });

  console.log('\n🔐 Issuing passport...');
  const passport = await vs.issuePassport({
    agentName: 'openai-coding-agent',
    owner:     'engineering@example.com',
    framework: 'openai',
  });
  console.log('   ✅ Issued!');
  console.log(`   DID:         ${passport.did}`);
  console.log(`   Trust Score: ${passport.trustScore}`);
  console.log(`   EU Risk:     ${passport.euRiskClass}`);
  console.log(`   Expires:     ${passport.expiresAt}`);

  console.log('\n🛡️  Verifying...');
  const ok = await vs.verify(passport.agentId);
  console.log(`   ${ok ? '✅ Verified' : '⛔ NOT verified'}`);

  console.log('\n🔍 Security scan...');
  const scan = await vs.scan(`
    const apiKey = "sk-1234";
    const result = eval(userInput);
  `);
  console.log(`   Threats: ${scan.threatCount}`);
  scan.threats.forEach(t => {
    const icon = t.severity === 'HIGH' ? '🔴' : '🟡';
    console.log(`   ${icon} [${t.severity}] Line ${t.line}: ${t.description}`);
  });

  console.log('\n⚖️  Compliance...');
  const c = await vs.checkCompliance(passport.agentId);
  console.log(`   EU AI Act: ${c.eu_ai_act.compliant ? '✅' : '❌'} (${c.eu_ai_act.risk_class})`);
  console.log(`   GDPR:      ${c.gdpr.compliant ? '✅' : '❌'}`);

  console.log('\n' + '='.repeat(55));
  console.log('  🚀 verisigilai.com  |  info@verisigilai.com');
  console.log('='.repeat(55));
}

main().catch(console.error);
